### ISSUE TITLE
DuplicateShellOnSteps: false positive with custom shell?

### ISSUE THREAD

**Issue by TWiStErRob** [TWiStErRob/net.twisterrob.ghlint#436]
### Version

ghlint 0.5.1-SNAPSHOT (ShellCheck build)

### Reproduction steps

1. File `.github/workflows/x.yml` with the following content:
   ```yaml
   on: push
   jobs:
     repro:
       runs-on: ubuntu-latest
       steps:
    steps:

      - name: "Checkout ${{ github.ref }} in ${{ github.repository }} repository."
        uses: actions/checkout@v4

      - name: "Output file changes"
        id: output-file-changes
        shell: echo {0}
        run: Hello
   ```

2. Run `ghlint` with the following arguments:
   ```shell
   ghlint --console .github/workflows/x.yml
   ```

### Expected behavior

Probably no flag.

### Actual behavior

> DuplicateShellOnSteps at .github/workflows/ci.yml:19:3-16: All (1) steps in Job[configuration] override shell as `echo {0}`, change the default shell on the job from `bash` to `echo {0}`, and remove shells from steps.
